common/fw/CMakeLists.txt

```diff
@@ -61,9 +61,9 @@ function(target_binary url version sha1 symbol ext)
   string(TOUPPER ${symbol} SYMBOL)
   string(REPLACE "." "," version_commas ${version})
   string(REPLACE "\\" "\\\\" binary_escaped "${binary}")
-  configure_file(fw.c.in  "${CMAKE_CURRENT_BINARY_DIR}/${symbol}.c" @ONLY)
-  configure_file(fw.h.in  "${CMAKE_CURRENT_BINARY_DIR}/${symbol}.h" @ONLY)
-  configure_file(fw.rc.in "${CMAKE_CURRENT_BINARY_DIR}/${symbol}.rc" @ONLY)
+  configure_file("${CMAKE_CURRENT_LIST_DIR}/fw.c.in"  "${CMAKE_CURRENT_BINARY_DIR}/${symbol}.c" @ONLY)
+  configure_file("${CMAKE_CURRENT_LIST_DIR}/fw.h.in"  "${CMAKE_CURRENT_BINARY_DIR}/${symbol}.h" @ONLY)
+  configure_file("${CMAKE_CURRENT_LIST_DIR}/fw.rc.in" "${CMAKE_CURRENT_BINARY_DIR}/${symbol}.rc" @ONLY)
```

Comment:
These changes and the one above shouldn't be needed.  This directory is added to the include search path by this line:
```cmake
target_include_directories(${PROJECT_NAME} PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>)
```
As long as `realsense2` correctly depends on the `fw` target this will happen automatically.  If you can't download the firmware for some reason, the variable enabling this gets turned off and then you will see an error related to these files not being found, but this is just a symptom.  I suspect your changes to force `BUILD_WITH_TM2` on should make this work for you.

Reply:
There is something about how this dependency is built from Android Studio. After retesting build I can confirm that changes to lines ## 64-65 are not required, but change to line # 7 is. W/o the change to this line build fails to resolve dependencies for this target